libraries/AP_Compass/Compass_learn.cpp

```diff
@@ -19,11 +19,11 @@ CompassLearn::CompassLearn(Compass &_compass) :
     compass(_compass)
 {
     gcs().send_text(MAV_SEVERITY_INFO, "CompassLearn: Initialised");
-    for (uint8_t i=0; i<compass.get_count(); i++) {
-        if (compass._state[i].use_for_yaw) {
+    for (Compass::Priority i(0); i<compass.get_count(); i++) {
```

Comment:
This lines scares me a bit.  Is it really going to learn all the compasses even those which have an index below the primary compasses's index?